packages/manager/src/managers/snippets/SnippetsManager.ts

```diff
@@ -0,0 +1,45 @@
+import * as t from "io-ts";
+import {
+	HookError,
+	Snippet,
+	SnippetReadHookData,
+} from "@slicemachine/plugin-kit";
+
+import { DecodeError } from "../../lib/DecodeError";
+import { assertPluginsInitialized } from "../../lib/assertPluginsInitialized";
+import { decodeHookResult } from "../../lib/decodeHookResult";
+
+import { BaseManager } from "../BaseManager";
+
+const snippetCodec = t.type({
```

Comment:
This would be an interesting feature. We discussed it on Slack in the past, and while I think it's a good idea, it presents some challenges.

If you have an idea of how this could work and how it would impact future plugins (think: non-JS languages), could you share a proposal?